tests/pyfunc/test_model_export_with_class_and_artifacts.py

```diff
@@ -1659,3 +1661,34 @@ def test_model_log_with_resources(tmp_path):
     pyfunc_model_path = _download_artifact_from_uri(pyfunc_model_uri)
     reloaded_model = Model.load(os.path.join(pyfunc_model_path, "MLmodel"))
     assert reloaded_model.resources == expected_resources
+
+
+def test_pyfunc_as_code_log_and_load(tmp_path):
+    with open("tests/pyfunc/pyfunc_sample_code.py") as file:
+        file_content = file.read()
+
+    temp_file = tmp_path / "model.py"
+    temp_file.write_text(file_content)
+
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            python_model=str(temp_file),
```

Comment:
Can we directly pass in `tests/pyfunc/pyfunc_sample_code.py`? 
Why do we need to write it to another file?